extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/IdentityProviderManagerBuilderCustomizerProducer.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.vertx.http.runtime.security;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.quarkus.security.spi.runtime.IdentityProviderManagerBuilder;
+import io.quarkus.vertx.runtime.VertxRecorder;
+
+@Singleton
+public class IdentityProviderManagerBuilderCustomizerProducer {
+
+    @Produces
+    @Singleton
+    public IdentityProviderManagerBuilder.Customizer produce() {
+        return new IdentityProviderManagerBuilder.Customizer() {
+            @Override
+            public void customize(IdentityProviderManagerBuilder<?> builder) {
+                builder.setEventLoopExecutor(VertxRecorder.getVertx().nettyEventLoopGroup());
```

Comment:
Note that this will dispatch back on 'an event loop thread' not 'the event loop thread', this makes the situation worse as now it may appear you are on an IO thread however you are not on your connections IO thread.

Reply:
Good point